topdown/tokens_test.go

```diff
@@ -1719,6 +1719,33 @@ func TestTopDownJWTDecodeVerify(t *testing.T) {
 			`{}`,
 			"",
 		},
+		{
+			"hs256-float-nbf",
+			`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ4eHgiLCJuYmYiOjEwMDAuMX0.8ab0xurlRs_glclA3Sm7OMQgwkQvE4HuLsfMOc4nVO8`,
+			`{"secret": "secret", "time": 2000000000000.1}`,
+			true,
+			`{"alg": "HS256", "typ": "JWT"}`,
+			`{"iss": "xxx", "nbf": 1000.1 }`,
+			"",
+		},
+		{
+			"hs256-float-exp-valid",
+			`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjMwMDAuMiwiaXNzIjoieHh4In0.XUen7GtDmICV3O1ngsoO-tQrjrXtOgJI06oGW0nQSIM`,
+			`{"secret": "secret", "time": 2000000000000.1}`,
+			true,
+			`{"alg": "HS256", "typ": "JWT"}`,
+			`{"iss": "xxx", "exp": 3000.2 }`,
+			"",
+		},
+		{
+			"hs256-float-exp-expired",
+			`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjMwMDAuMiwiaXNzIjoieHh4In0.XUen7GtDmICV3O1ngsoO-tQrjrXtOgJI06oGW0nQSIM`,
+			`{"secret": "secret", "time": 4000000000000.1}`,
+			false,
+			`{}`,
+			`{}`,
+			"",
+		},
```

Comment:
Could you add in a couple more test cases, I'd like to see:

* When they're equal (to ensure we catch any issues with comparing floats, and any regressions if someone messes with the "if" checks and the compare results), for both `exp` and `nbf`
* A negative case for the `nbf` with floats